### PR TITLE
Refactor IdempotencyPropertyTest into CorrectnessTest and PropertyTest

### PR DESCRIPTION
**Pull request by MasseGuillaume**
PropertyTest contains all the logic to drive the property test.
* checks for regression
* progressbar
* todo file for failures

CorrectnessTest contains the meat for the property test.

CorrectnessTest

### CODE REVIEW

File: tests/slow/src/test/scala/org/scalafmt/internal/SyntaxAnalysis.scala [hunk 1]
```diff
@@ -1,34 +1,16 @@
...
-import scala.collection.JavaConverters._
-import scala.collection.mutable
-import scala.meta._
 import scala.util.control.NonFatal
 
-import java.util.concurrent.CopyOnWriteArrayList
-import java.util.concurrent.atomic.AtomicInteger
-
 object SyntaxAnalysis {
-
```
Comment (MasseGuillaume) [inline tests/slow/src/test/scala/org/scalafmt/internal/SyntaxAnalysis.scala:15]: We don't need to aggregate test value.

File: tests/slow/coverage-correctness.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+scala-js/test-suite/js/src/test/resources/SourceMapTestTemplate.scala
```
Comment (MasseGuillaume) [inline tests/slow/coverage-correctness.txt]: @olafurpg :(

File: tests/slow/src/test/scala/org/scalafmt/internal/PropertyTest.scala [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+  val previouslyFailed: Set[File] =
+    if (Files.exists(coverageFile)) {
+      val input = new String(Files.readAllBytes(coverageFile))
+      input.split(nl).filterNot(_ == "").map(f => new File(prefix + f)).toSet
+    } else {
+      Set()
+    }
+
+  def check(file: Input.File): Either[String, Unit]
```
Comment (olafurpg) [inline tests/slow/src/test/scala/org/scalafmt/internal/PropertyTest.scala]: Can we move the abstract method to the top and add a comment explaining what `Either[String, Unit]`  means

File: tests/slow/src/test/scala/org/scalafmt/internal/CorrectnessTest.scala [hunk 4]
```diff
@@ -0,0 +1,33 @@
+package org.scalafmt.internal
+
+import scala.meta._
+import org.scalafmt.Options
+
+import scala.meta.testkit.StructurallyEqual
+
+object CorrectnessTest extends PropertyTest("correctness") {
```
Comment (olafurpg) [inline tests/slow/src/test/scala/org/scalafmt/internal/CorrectnessTest.scala]: "correctness" can be interpreted in many ways. Throwing away comments can be treated as "incorrect". "idempotent AST" leaves less room for interpretation, IMO. Either way, would be good to write a one line summary defining explicitly what property is being tested here.

**Review by olafurpg — CHANGES_REQUESTED** [review]

**Review by olafurpg — APPROVED** [review]
LGTM 👍

**Pull request merged**